src/openapi.json

```diff
@@ -0,0 +1,191 @@
+{
+  "openapi": "3.0.3",
+  "info": {
+    "title": "Philanthropy Data Commons API",
+    "description": "An API for a common data platform to make the process of submitting data requests to funders less burdensome for organizations seeking grants.",
+    "version": "0.0.4"
+  },
+  "components": {
+    "parameters": {
```

Comment:
Looking at the OpenAPI spec ma few thoughts:

1. The example given in the OpenAPI documentation actually explicitly flags components/parameters as being *operation* parameters -- e.g. for things that truly are "meta" parameters that are used to provide decorative functionality across multiple endpoints (e.g. `limit` or `offset`).

2. That said, I appreciate there is DRY value in specifying things that are simply reused, and that was the original intent.

So that brings me to a question / suggestion!

I think having `opportunityId` defined in three ways is not ideal (e.g. opportunityId, opportunityIdQueryOptional, opportunityIdQueryRequired); it looks like MAYBE you could define it without specifying required in the `components` and then in the actual API you could say:

```
        "parameters": [
          {
            "$ref": "#/components/parameters/opportunityId",
            "required": false,
          }
        ],
```

I haven't tested this but if it works that seems way nicer (and less repetitive)

Reply:
This is a great suggestion and makes perfect sense to me, I am trying it out and my IDE plugin is complaining when I remove either `in` or `path` but let me see if the swagger.io website validator complains.

Reply:
Darn. Both the "OpenAPI (Swagger) Editor" codium extension and also https://editor.swagger.io/ say "no dice" to removing either of `in` or `path`.

Reply:
Just to make sure that's not a bug in the two validators, I see those fields marked "required" in the spec too, at https://swagger.io/specification/#parameterObject

At the moment, some of these are only used once, so it does not make sense to define them up top until there are two uses. I guess I anticipate multiple uses of these but that could be deferred to later. Or perhaps it is simpler/clearer to always define parameters up top in the common parameters area rather than some defined here while others are defined there.

Reply:
It is OK to omit the `required` property when the parameter location is `query`, but then it causes the value to be interpreted `false` even when an override to `true` is attempted in the context of the specific path.

Reply:
So I think our best bet may be to just define these things inline rather than in a reusable place -- even when a parameter is "reused" it will generally not be *identical* (just similar in several ways) -- e.g. in one case the opportunityId would be what is being accessed, in another it would be the opportunityId for what is being deleted, in another it would be the opportunityId for what is being updated, etc.

Would that solve the problem?

I do think this piece of the swagger spec is intended for operational parameters that are truly reused (e.g. parameters like `orderBy` or `limit` or `offset` or `auth`)

Reply:
In the narrowest sense, I think there will probably be re-use of some optional query parameters, although I agree it's not there now so not needed now. In the general sense, the parameters regarding opportunityId (whether required or not, whether in this spot or that spot) really are all referring to the same opportunityId. An opportunityId by any other name would smell as sweet. So the thing referred to, the opportunityId or the opportunity itself, is truly identical, but the operation attempted may require it here or there or not at all. Those aren't properties of opportunities but properties of the API spec.

But yes, for the moment, I think they should be inline, because they are not really re-used yet.